Modules/DialogManagement/src/Game.DialogManagement.Domain/Repositories/IDialogueRepository.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using Game.DialogManagement.Domain.Data.Entities;
+
+namespace Game.DialogManagement.Domain.Repositories
+{
+    /// <summary>
+    /// The interface implemented by anything that wants to store dialogues.
+    /// </summary>
+    public interface IDialogueRepository
+    {
+        /// <summary>
+        /// Gets or creates a dialogue.
+        /// </summary>
+        /// <param name="playerId">The id of the player conversing.</param>
+        /// <param name="npcId">The id of the npc they're conversing with</param>
+        /// <returns>The fetched/created dialogue.</returns>
+        Dialogue GetOrCreate(Guid playerId, Guid npcId);
```

Comment:
W metodzie _GetOrCreate_ Rozdzieliłbym Get oraz Create, i sprawdzał na poziomie kontrolerów, bądź jeżeli istnieje taka możliwość to aplikacji, choć chyba na poziomie kontrolerów będzie lepiej i łatwiej, czy get coś zwrócił, jeżeli nie to utworzyłbym dialog, w celu rozróżnienia operacji Query - Get od Command - Create, powodując się na zasady _CQRS_.